infrastructure-playbooks/shrink-mds.yml

```diff
@@ -0,0 +1,110 @@
+---
+# This playbook removes the Ceph MDS from your cluster.
+#
+# Use it like this:
+# ansible-playbook shrink-mds.yml -e mds_to_kill=ceph-mds01
+#     Prompts for confirmation to shrink, defaults to no and
+#     doesn't shrink the cluster. yes shrinks the cluster.
+#
+# ansible-playbook -e ireallymeanit=yes|no shrink-mds.yml
+#     Overrides the prompt using -e option. Can be used in
+#     automation scripts to avoid interactive prompt.
+- name: gather facts and check the init system
+  hosts:
+    - "{{ mon_group_name | default('mons') }}"
+    - "{{ mds_group_name | default('mdss') }}"
+  become: true
+  tasks:
+    - debug: msg="gather facts on all Ceph hosts for following reference"
+
+- name: perform checks, remove mds and print cluster health
+  hosts: localhost
+  become: true
+  vars_prompt:
+    - name: ireallymeanit
+      prompt: Are you sure you want to shrink the cluster?
+      default: 'no'
+      private: no
+  vars:
+    mds_group_name: mdss
+  pre_tasks:
+    - name: exit playbook, if no mds was given
+      when: mds_to_kill is not defined
+      fail:
+        msg: >
+          mds_to_kill must be declared.
+          Exiting shrink-cluster playbook, no MDS was removed. On the command
+          line when invoking the playbook, you can use
+          "-e mds_to_kill=ceph-mds1" argument.  You can only remove a single
+          MDS each time the playbook runs."
+
+    - name: exit playbook, if the mds is not part of the inventory
+      when: mds_to_kill not in groups[mds_group_name]
+      fail:
+        msg: "It seems that the host given is not part of your inventory,
+              please make sure it is."
+
+    - name: exit playbook, if user did not mean to shrink cluster
+      when: ireallymeanit != 'yes'
+      fail:
+        msg: "Exiting shrink-mds playbook, no mds was removed.
+           To shrink the cluster, either say 'yes' on the prompt or
+           or use `-e ireallymeanit=yes` on the command line when
+           invoking the playbook"
+
+    - import_role:
```

Comment:
you should move this task before any other task here

Reply:
Done.